src/custom-elements.js

```diff
@@ -28,18 +28,21 @@ if (!priorCustomElements || priorCustomElements['forcePolyfill']) {
   PatchNode(internals);
   PatchElement(internals);
 
+  // The main document is always associated with the registry.
+  document.__CE_hasRegistry = true;
+
   /** @type {!CustomElementRegistry} */
   const customElements = new CustomElementRegistry(internals);
 
   // If `customElements.documentReady` exists and is a Promise, prevent calls to
   // define from causing upgrades until `documentReady` has resolved.
-  if (priorCustomElements) {
+  if (priorCustomElements && priorCustomElements['documentReady']) {
     const documentReady = priorCustomElements['documentReady'];
-    if (documentReady && documentReady.then instanceof Function) {
+    if (documentReady.then instanceof Function) {
```

Comment:
Sorry I didn't notice this comment earlier, I put this change in a PR #54, PTAL. I think the API matches what you've described:
- If the callback is defined, `customElements.define` will call the callback with a function; this function can be used to trigger the document walk.
```javascript
window.customElements = {
  flushCallback: function(doFlush) {
    otherThing.doFlush = doFlush;
  },
};
// ... custom elements polyfill runs here ...
customElements.define('element-a', class extends HTMLElement {});
// Synchronously during the above call, `flushCallback` is called with a function.
```
- If multiple calls to `customElements.define` occur, only the first will cause the callback to be called (with the function).
```javascript
// .. continuing from above.
customElements.define('element-b', class extends HTMLElement {});
//`flushCallback` is *not* called.
```
- If the function *given* to the callback is called to trigger a document walk, the *next* `customElements.define` call will again call the callback (giving the document-walk-triggering function).
```javascript
// .. continuing from above.
otherThing.doFlush();
// ^^ This causes the document to be walked and promises given by `whenDefined`
// for 'element-a' and 'element-b' to be resolved.
customElements.define('element-c', class extends HTMLElement {});
// `flushCallback` is called with the upgrade-triggering function again.
```
I don't think we discussed how to disable this, so the way I handle this is to provide a function called `customElements.polyfillSetFlushCallback` that you can use to add / change / remove the callback. Giving it a function starts using that function as the callback; giving it `undefined` automatically flushes anything that was delayed and stops using the last callback. Similar to `forcePolyfill`, if a property named `flushCallback` exists on a global object called `customElements` when the polyfill starts up, it will use this function initially.
```javascript
// .. continuing from above.
customElements.polyfillSetFlushCallback(undefined);
// ^^ The document is walked and promises for 'element-c' are resolved.
// Later calls to `customElements.define` do *not* call the callback and, instead,
// they synchronously walk the document and resolve `whenDefined` promises.
```